mp/src/game/client/momentum/ui/HUD/hud_stickybombs.cpp

```diff
@@ -1,20 +1,62 @@
 #include "cbase.h"
 
+#include <vgui_controls/AnimationController.h>
 #include <vgui_controls/EditablePanel.h>
-#include <vgui_controls/Label.h>
 
 #include "c_mom_player.h"
 #include "hudelement.h"
 #include "iclientmode.h"
 #include "mom_system_gamemode.h"
 #include "weapon/weapon_mom_stickybomblauncher.h"
+#include "weapon/weapon_shareddefs.h"
 
 #include "tier0/memdbgon.h"
 
 using namespace vgui;
 
 static MAKE_TOGGLE_CONVAR(mom_hud_sj_stickycount_enable, "1", FCVAR_ARCHIVE, "Toggles the stickybomb counter.\n");
-static MAKE_TOGGLE_CONVAR(mom_hud_sj_stickycount_autohide, "0", FCVAR_ARCHIVE, "Toggles automatically hiding the stickybomb counter at 0 stickies. 0 = OFF, 1 = ON\n");
+
+class Stickybox : public Panel
+{
+    DECLARE_CLASS_SIMPLE(Stickybox, Panel);
+
+  public:
+    Stickybox(Panel *pParent);
+
+    bool HasAnimationStarted() const { return m_bAnimationStarted; }
+
+    void StartAnimation(Color start, Color target);
+    void StopAnimation();
+
+    void OnThink() override;
+
+  private:
+    CPanelAnimationVar(Color, m_StickyColor, "StickyColor", "BlackHO");
+
+    bool m_bAnimationStarted;
+};
+
+Stickybox::Stickybox(Panel *pParent) : BaseClass(pParent, "Stickybox"), m_bAnimationStarted(false)
+{}
+
+void Stickybox::StartAnimation(Color start, Color target)
+{
+    m_StickyColor = start;
+    g_pClientMode->GetViewportAnimationController()->RunAnimationCommand(this, "StickyColor", target, 0.0f, C_MomStickybomb::GetArmTime(), AnimationController::INTERPOLATOR_ACCEL);
```

Comment:
Regarding your [comment](https://github.com/momentum-mod/game/pull/935#issuecomment-694706920):
> Is there a better way of doing this?

Yes, but involves getting a little bit dirty.

AnimationController has only two ways to run animations, `StartAnimationSequence` and `RunAnimationCommand`. What we want is a hybrid between the two; running a specific sequence, but for a specific panel, while ignoring the actual panel name defined in the animation. I'm genuinely surprised this isn't possible, as `StartAnimationSequence` has an overload definition that takes a panel, but is supposed to be the *starting point for finding the named panel in the sequence*. Thanks Valve. 🙄 

So what I want you to do is make a `StartAnimationSequenceForPanel(Panel *pPanel, const char *pSequence)` method. This will effectively do the best of both worlds, so it'll look like:
```cpp
bool AnimationController::StartAnimationSequenceForPanel(Panel *pPanel, const char *sequenceName)
{
    Assert( pPanel);

	if (m_bAutoReloadScript)
	{
		// Reload the script files
		ReloadScriptFile();
	}

	// lookup the symbol for the name
	UtlSymId_t seqName = g_ScriptSymbols.Find(sequenceName);
	if (seqName == UTL_INVAL_SYMBOL)
		return false;

	// Msg("Starting animation sequence %s\n", sequenceName);

	// remove the existing command from the queue
	RemoveQueuedAnimationCommandsForPanel(seqName, pPanel);

	// look through for the sequence
	int i;
	for (i = 0; i < m_Sequences.Count(); i++)
	{
		if (m_Sequences[i].name == seqName)
			break;
	}
	if (i >= m_Sequences.Count())
		return false;

	// execute the sequence
	for (int cmdIndex = 0; cmdIndex < m_Sequences[i].cmdList.Count(); cmdIndex++)
	{
		ExecAnimationCommandForPanel(seqName, m_Sequences[i].cmdList[cmdIndex], pPanel);
	}

	return true;	
}

```

The goal here is to make it so that instead of *searching* for the target panel (which is typically defined in the HudAnimations.txt file), we're just *giving* it to the methods. So then you'll need to copy the implementation of `ExecAnimationCommand` and make it automatically use the `StartCmd_Animate` that can directly use the panel pointer of `pPanel`.

You may also need to define a `RemoveQueuedAnimationCommandsForPanel` that does the same as `RemoveQueuedAnimationCommands` but for a given panel (see how it searches for the target panel again?).

Reply:
Oh and try to use `vgui::GetAnimationController` instead of having to fetch it from the clientmode. If animations break, feel free to use the clientmode one, but for the most part, the vgui namespace one should be good to use (and less verbose!).